app/ts/utils/typescript.ts

```diff
@@ -29,3 +29,8 @@ export function getWithDefault<Key, Value>(map: Map<Key, Value>, key: Key, defau
 	if (previousValue === undefined) return defaultValue
 	return previousValue
 }
+
+type Split<T> = { [K in keyof T]: { [P in K]: T[P] } }[keyof T] | Record<PropertyKey, never>
+export function modifyObject<T extends object>(original: T, subObject: NoInfer<Split<T>>): T {
```

Comment:
I may be missing some test cases for this but looks to me like `Partial<T>` would already be sufficient for type `subObject`?

```ts
export function modifyObject<T extends object>(original: T, subObject: Partial<T>) {
	return {...original, subObject }
}
```